src/include/planner/operator/logical_gds_call.h

```diff
@@ -10,28 +10,38 @@ class LogicalGDSCall final : public LogicalOperator {
     static constexpr LogicalOperatorType operatorType_ = LogicalOperatorType::GDS_CALL;
 
 public:
-    explicit LogicalGDSCall(binder::BoundGDSCallInfo info,
-        std::vector<std::shared_ptr<LogicalOperator>> nodeMaskRoots)
-        : LogicalOperator{operatorType_}, info{std::move(info)},
-          nodeMaskRoots{std::move(nodeMaskRoots)} {}
+    explicit LogicalGDSCall(binder::BoundGDSCallInfo info)
+        : LogicalOperator{operatorType_}, info{std::move(info)} {}
+
+    // Note that this is a special interface for adding node mask roots as children.
+    // We shouldn't have other cases where this operator has any children.
+    void addDummySinkAsMaskChild(std::shared_ptr<LogicalOperator> child) {
```

Comment:
Why not use addChild in the base LogicalOperator?

Reply:
hmm I don't think we have that in the base LogicalOperator? Should I add that? I think it's better not to propagate this interface to the base as the usage is very specific to few operators now?